DOCSP-41772-export-schema (#731)
mongodb/docs-compass: source/includes/schema-export-format-standard.rst

Standard format schema objects contain the following fields:

.. list-table::
   :header-rows: 1
   :widths: 35 25 40
 
   * - Property
     - Data type
     - Description

   * - ``type``
     - string or array
     - JSON type of this data type. For details, see the `official JSON 
       docs <https://json-schema.org/draft/2020-12/json-schema-validation#name-type>`_.

   * - ``required``
     - array of strings
     - Fields that must appear in the schema. For details, see the 
       `official JSON docs <https://json-schema.org/draft/2020-12/json-schema-validation#name-required>`_.

   * - ``properties``

>      - document

Reviewer: - object (where the keys are property names and values are subschemas)

>      - document

Reviewer: ```suggestion
     - object
```

Author: Updated.

>      - Properties for each field. Each subdocument may contain fields 
>        like ``type``. For details, see the `official JSON docs 
>        <https://json-schema.org/draft/2020-12/json-schema-core#section-10.3.2.1>`_.

Reviewer: ```suggestion
     - Properties for each field.  Keys are property names and values are subschemas. For details, see the `official JSON docs 
       <https://json-schema.org/draft/2020-12/json-schema-core#section-10.3.2.1>`_.
```

Author: Updated.

 
   * - ``items``
     - document

>      - Metadata about elements in array fields. Each subdocument may contain

Author: [question]

Should this one have a similar description to ``properties`` above, like "Keys are property names and values are subschemas."? Thanks!

Reviewer: Yes, I was just about to get to it! Items are similar to properties, but this time it is a subschema (directly).

Standard format schema objects contain the following fields:

.. list-table::
   :header-rows: 1
   :widths: 35 25 40
 
   * - Property
     - Data type
     - Description

   * - ``type``
     - string or array

>      - |bson| type of this data type. For details, see the `official JSON 

Reviewer: So this property would contain JSON schema type, not BSON type. We'll be converting BSON type to standard JSON Schema type (when possible), or referring to a relaxed extended EJSON definition when not (https://www.mongodb.com/docs/manual/reference/mongodb-extended-json).

Reviewer: Example simple type:
```
"year": {
  "type": "integer"
 }
```
vs 
```
"year": {
  "bsonType": "int"
 }
```

Example complex type (can't be easily described via JSON type):
```
"_id": {
   "$ref": "#/$defs/ObjectId"
},
```
vs
```
"_id": {
  "bsonType": "objectId"
},
```

Author: Thanks! Updated.

       docs <https://json-schema.org/draft/2020-12/json-schema-validation#name-type>`_.

   * - ``required``

>      - string or array

Reviewer: ```suggestion
     - array of strings
```

Reviewer: This one must be an array

     - Fields that must appear in the schema. For details, see the 
       `official JSON docs <https://json-schema.org/draft/2020-12/json-schema-validation#name-required>`_.

   * - ``properties``
     - document
     - Properties for each field, such as ``type`` and ``description``.
       For details, see the `official JSON docs 
       <https://json-schema.org/draft/2020-12/json-schema-core#section-10.3.2.1>`_.
 
   * - ``properties.<field>.type``
     - string
     - The data type of the ``<field>``.

   * - ``properties.<field>.description``
     - string
     - Human-readable description of the ``<field>``. 

   * - ``properties.<field>.items``
     - document
     - Types of elements in array fields. For details, see the `official

>        JSON docs <https://json-schema.org/draft/2020-12/json-schema-core#section-10.3.1.2>`_.

Reviewer: This part doesn't make sense. 
Properties exist for `type: object`, they are an object where keys are the fields and values are subschemas (we don't need to go into details with `properties.<field>.type`, because in fact there can be many levels of sub-schemas). Items exist for `type: array`. Items are subschema.
I also need to note that this is not an exhaustive list. It might make sense to list the most common fields, but we need to highlight that we're following a the [JSON Schema Draft 2020-12](https://json-schema.org/draft/2020-12) specification.
